pysat/utils/files.py

```diff
@@ -363,17 +411,17 @@ def construct_searchstring_from_format(format_str, wildcard=False):
             out_dict['keys'].append(snip[1])
 
             # Try and determine formatting width
-            temp = re.findall(r'\d+', snip[2])
+            fwidths = re.findall(r'\d+', snip[2])
 
-            if temp:
+            if fwidths:
                 # There are items, try and grab width
-                for i in temp:
+                for fwidth in fwidths:
                     # Make sure there is truly something there
-                    if i != 0:
+                    if fwidth != 0:
                         # Store length and add to the search string
-                        out_dict['lengths'].append(np.int64(i))
+                        out_dict['lengths'].append(np.int64(fwidth))
                         if not wildcard:
-                            out_dict['search_string'] += '?' * np.int64(i)
+                            out_dict['search_string'] += '?' * np.int64(fwidth)
```

Comment:
```suggestion
                            out_dict['search_string'] += '?' * out_dict['lengths'][-1]
```

To avoid calling the same operation twice.